src/lib/Name.hs

```diff
@@ -173,9 +173,11 @@ class (SinkableB scopeFrag, BindsNames scopeFrag) => OutFrag (scopeFrag :: S ->
 class HasScope scope => OutMap scope where
   emptyOutMap :: scope VoidS
 
-class (OutFrag scopeFrag, OutMap scope)
-      => ExtOutMap (scope :: S -> *) (scopeFrag :: S -> S -> *) where
-  extendOutMap :: Distinct l => scope n -> scopeFrag n l -> scope l
+class OutMap env => ExtOutMap (env :: S -> *) (frag :: S -> S -> *) where
```

Comment:
It's because we also want to use `ExtOutMap` for the sub-fragments, but those sub-fragments themselves are often inconvenient to make `OutFrag`s. For example, we often want to treat a single `Decl` as a sub-fragment, but a `Decl` is not an `OutFrag`, since there's no way to concatenate two `Decls` while still getting another one.